### PR TITLE
Increase code coverage createXXXFromJson methods in RealmJsonTests.

### PR DESCRIPTION
**Pull request by dalinaum**
* createAllFromJson(Class<E>, String)
 * createObjectFromJson(Class<E>, InputStream)
 * createOrUpdateObjectFromJson(Class<E>, InputStream)
 * createOrUpdateObjectFromJson(Class<E>, JSONObject)
 * createOrUpdateObjectFromJson(Class<E>, String)

In progress, I've rearranged some methods. Without rearrangements, I cannot figure out what methods are not covered.

### CODE REVIEW

File: realm/realm-library/src/androidTest/java/io/realm/RealmJsonTests.java [hunk 1]
```diff
@@ -592,6 +592,38 @@ public void createObjectFromJson_streamNoValues() throws IOException {
...
     }
 
+    @Test
+    public void createObjectFromJson_streamNullClass() throws IOException {
+        InputStream in = TestHelper.loadJsonFromAssets(context, "array.json");
+        realm.beginTransaction();
+        AnnotationTypes result = realm.createObjectFromJson(null, in);
+        realm.cancelTransaction();
+        in.close();
+        assertNull(result);
```
Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/RealmJsonTests.java]: Shouldn't we assert be we cancel the transaction?

File: realm/realm-library/src/androidTest/java/io/realm/RealmJsonTests.java [hunk 2]
```diff
@@ -592,6 +592,38 @@ public void createObjectFromJson_streamNoValues() throws IOException {
...
+            in.close();
+        }
+    }
+
+    @Test
+    public void createObjectFromJson_streamNullInputStream() throws IOException  {
+        realm.beginTransaction();
+        AnnotationTypes result = realm.createObjectFromJson(AnnotationTypes.class, (InputStream) null);
+        realm.cancelTransaction();
+        assertNull(result);
```
Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/RealmJsonTests.java]: same

**Comment by dalinaum** [comment]
Realm: 81%, (278 of 1,479) ->  85% (226 of 1,479)

Before:
screen shot 2016-04-13 at 4 08 56 pm

After:
screen shot 2016-04-13 at 4 09 14 pm

There are only 1 missed instruction and some missed branches but most instructions of createXXXFromJson are covered.

**Comment by dalinaum** [comment]
Please review this @realm/java

**Comment by kneth** [comment]
Looks fine to me

**Comment by cmelchior** [comment]
:+1:

**Pull request merged**